.github/workflows/reusable_testing.yml

```diff
@@ -45,15 +45,15 @@ jobs:
           go clean -testcache && go test -v
           kill $TNT_PID
 
-# TODO(ylobankov): Uncomment this when tarantool/go-tarantool#115 is resolved.
-#      - name: Run queue tests
-#        working-directory: ./queue
-#        run: |
-#          mkdir snap xlog
-#          tarantoolctl rocks install queue 1.1.0
-#          TNT_PID=$(tarantool ./config.lua > tarantool.log 2>&1 & echo $!)
-#          go clean -testcache && go test -v
-#          kill $TNT_PID
+      - name: Run queue tests
+        working-directory: ./queue
+        run: |
+          mkdir snap xlog
+          tarantoolctl rocks install queue 1.1.0
+          TNT_PID=$(tarantool ./config.lua > tarantool.log 2>&1 & echo $!)
+          go clean -testcache && go test -v
+          kill $TNT_PID
+        if: ${{ !startsWith(env.TNT_VERSION, 'Tarantool 1.10') }}
```

Comment:
What's the deal with 1.10?

Reply:
I've clashed with some problem on calling `box.cfg`  in tarantool 1.10. That looks like some problem with initializing
collations. See the logs:
```
2022-01-26 10:13:45.506 [1706] main/101/config.lua C> Tarantool 1.10.12-22-g73a90f94c
2022-01-26 10:13:45.507 [1706] main/101/config.lua C> log level 5
2022-01-26 10:13:45.507 [1706] main/101/config.lua I> mapping 268435456 bytes for memtx tuple arena...
2022-01-26 10:13:45.507 [1706] main/101/config.lua I> Actual slab_alloc_factor calculated on the basis of desired slab_alloc_factor = 1.044274
2022-01-26 10:13:45.507 [1706] main/101/config.lua I> mapping 134217728 bytes for vinyl tuple arena...
2022-01-26 10:13:45.508 [1706] main/101/config.lua I> instance uuid a5fec003-cf8a-45ed-9ff5-3931e118b2cb
2022-01-26 10:13:45.508 [1706] iproto/101/main I> binary: bound to [::]:3013
2022-01-26 10:13:45.508 [1706] main/101/config.lua I> initializing an empty data directory
2022-01-26 10:13:45.509 [1706] main/101/config.lua alter.cc:2380 E> ER_CANT_CREATE_COLLATION: Failed to initialize collation: unknown collation type.
2022-01-26 10:13:45.509 [1706] main/101/config.lua F> can't initialize storage: Failed to initialize collation: unknown collation type.
```
I've noticed that in some steps there is the same problem and these steps constraint test running on tarantool 1.10. So I've done the same until we discuss it and maybe fix this problem.

Reply:
This is weird. How collations related to queue test? I don't found any explicit collation specifications both here and in tarantool/queue.

Reply:
Maybe you've tried to start Tarantool 1.10 over Tarantool 2.x snapshots?

Reply:
Idk, I thought the same, but I've cleared the directory before start. The effect is the same.

Reply:
Well, seems like this was really the issue.
Snapshots from 2.x:
```
tarantool ./config.lua
Deprecated option snap_dir, please use memtx_dir instead
2022-01-26 12:58:04.606 [23294] main/101/config.lua C> Tarantool Enterprise 1.10.11-0-gf0b0e7ecf-r423
2022-01-26 12:58:04.607 [23294] main/101/config.lua C> log level 5
2022-01-26 12:58:04.607 [23294] main/101/config.lua I> mapping 268435456 bytes for memtx tuple arena...
2022-01-26 12:58:04.607 [23294] main/101/config.lua I> Actual slab_alloc_factor calculated on the basis of desired slab_alloc_factor = 1.044274
2022-01-26 12:58:04.608 [23294] main/101/config.lua I> mapping 134217728 bytes for vinyl tuple arena...
2022-01-26 12:58:04.610 [23294] main/101/config.lua I> instance uuid 2599ce5f-b33c-44eb-acf2-a8cadbf8f3d4
2022-01-26 12:58:04.610 [23294] main/101/config.lua I> instance vclock {1: 49}
2022-01-26 12:58:04.611 [23294] iproto/101/main I> binary: bound to 0.0.0.0:3013
2022-01-26 12:58:04.611 [23294] main/101/config.lua I> recovery start
2022-01-26 12:58:04.611 [23294] main/101/config.lua I> recovering from `snap/00000000000000000000.snap'
2022-01-26 12:58:04.612 [23294] main/101/config.lua I> cluster uuid 406c1498-783a-495d-b8a3-8601ad42240b
2022-01-26 12:58:04.612 [23294] main/101/config.lua alter.cc:2380 E> ER_CANT_CREATE_COLLATION: Failed to initialize collation: unknown collation type.
2022-01-26 12:58:04.612 [23294] main/101/config.lua F> can't initialize storage: Failed to initialize collation: unknown collation type.
```

Clean up dirs
```
rm -rf ./snap
rm -rf ./xlog
mkdir snap
mkdir xlog
```

```
tarantool ./config.lua 
Deprecated option snap_dir, please use memtx_dir instead
2022-01-26 12:58:27.969 [23309] main/101/config.lua C> Tarantool Enterprise 1.10.11-0-gf0b0e7ecf-r423
2022-01-26 12:58:27.969 [23309] main/101/config.lua C> log level 5
2022-01-26 12:58:27.969 [23309] main/101/config.lua I> mapping 268435456 bytes for memtx tuple arena...
2022-01-26 12:58:27.970 [23309] main/101/config.lua I> Actual slab_alloc_factor calculated on the basis of desired slab_alloc_factor = 1.044274
2022-01-26 12:58:27.970 [23309] main/101/config.lua I> mapping 134217728 bytes for vinyl tuple arena...
2022-01-26 12:58:27.971 [23309] main/101/config.lua I> instance uuid d82f1274-2982-4c55-9fec-bba08fdd030a
2022-01-26 12:58:27.971 [23309] iproto/101/main I> binary: bound to 0.0.0.0:3013
2022-01-26 12:58:27.971 [23309] main/101/config.lua I> initializing an empty data directory
2022-01-26 12:58:27.974 [23309] main/101/config.lua I> assigned id 1 to replica d82f1274-2982-4c55-9fec-bba08fdd030a
2022-01-26 12:58:27.974 [23309] main/101/config.lua I> cluster uuid 0d32e3f4-0f83-421c-b44a-a8715b0ba93f
2022-01-26 12:58:27.974 [23309] snapshot/101/main I> saving snapshot `snap/00000000000000000000.snap.inprogress'
2022-01-26 12:58:27.976 [23309] snapshot/101/main I> done
2022-01-26 12:58:27.977 [23309] main/101/config.lua I> ready to accept requests
2022-01-26 12:58:27.977 [23309] main/107/checkpoint_daemon I> started
2022-01-26 12:58:27.977 [23309] main/107/checkpoint_daemon I> scheduled the next snapshot at Wed Jan 26 14:56:39 2022
2022-01-26 12:58:27.987 [23309] main schema.lua:1817 E> ER_NO_SUCH_SPACE: Space '_trigger' does not exist
2022-01-26 12:58:27.987 [23309] main F> fatal error, exiting the event loop
```
Commented row with `_trigger`
```
tarantool ./config.lua 
Deprecated option snap_dir, please use memtx_dir instead
2022-01-26 12:59:15.458 [23351] main/101/config.lua C> Tarantool Enterprise 1.10.11-0-gf0b0e7ecf-r423
2022-01-26 12:59:15.458 [23351] main/101/config.lua C> log level 5
2022-01-26 12:59:15.459 [23351] main/101/config.lua I> mapping 268435456 bytes for memtx tuple arena...
2022-01-26 12:59:15.459 [23351] main/101/config.lua I> Actual slab_alloc_factor calculated on the basis of desired slab_alloc_factor = 1.044274
2022-01-26 12:59:15.459 [23351] main/101/config.lua I> mapping 134217728 bytes for vinyl tuple arena...
2022-01-26 12:59:15.459 [23351] main/101/config.lua I> instance uuid d82f1274-2982-4c55-9fec-bba08fdd030a
2022-01-26 12:59:15.459 [23351] main/101/config.lua I> instance vclock {1: 36}
2022-01-26 12:59:15.459 [23351] iproto/101/main I> binary: bound to 0.0.0.0:3013
2022-01-26 12:59:15.460 [23351] main/101/config.lua I> recovery start
2022-01-26 12:59:15.460 [23351] main/101/config.lua I> recovering from `snap/00000000000000000000.snap'
2022-01-26 12:59:15.460 [23351] main/101/config.lua I> cluster uuid 0d32e3f4-0f83-421c-b44a-a8715b0ba93f
2022-01-26 12:59:15.461 [23351] main/101/config.lua I> assigned id 1 to replica d82f1274-2982-4c55-9fec-bba08fdd030a
2022-01-26 12:59:15.461 [23351] main/101/config.lua I> recover from `xlog/00000000000000000000.xlog'
2022-01-26 12:59:15.461 [23351] main/101/config.lua I> done `xlog/00000000000000000000.xlog'
2022-01-26 12:59:15.461 [23351] main/101/config.lua I> recover from `xlog/00000000000000000036.xlog'
2022-01-26 12:59:15.461 [23351] main/101/config.lua I> done `xlog/00000000000000000036.xlog'
2022-01-26 12:59:15.461 [23351] main/101/config.lua I> ready to accept requests
2022-01-26 12:59:15.461 [23351] main/101/config.lua C> leaving orphan mode
2022-01-26 12:59:15.461 [23351] main/107/checkpoint_daemon I> started
2022-01-26 12:59:15.462 [23351] main/107/checkpoint_daemon I> scheduled the next snapshot at Wed Jan 26 14:02:39 2022
2022-01-26 12:59:15.462 [23351] main C> entering the event loop
```

So the only error I've got for Tarantool 1.10 started from clean work dirs is unknown `_trigger` space.

Reply:
I see you removed `Tarantool != 1.10` constraint in "testing" workflow, but you're forgot to remove it in "reusable_testing"

Reply:
Well, your approach to add `_trigger` space if it doesn't exists works, but I don't think it's correct. `_trigger` is a space added in 2.x for internal needs of SQL: https://github.com/tarantool/tarantool/commit/7cb3d2cfe6d2478dcf4a87527eb6e580ade90d6b , so it is unsurprising that we don't have `_trigger` in 1.10. But AFAIK we don't use sql in queue and this grants are useless for us (and for tests), so we can just remove `box.schema.user.grant('test','read','space','_trigger')`

Reply:
That sounds reasonable, but if I'm removing this grant I get the next output in tests:
```sh
=== RUN   TestFifoQueue
    queue_test.go:46: Failed drop queue: Read access to space '_trigger' is denied for user 'test' (0x2a)
--- FAIL: TestFifoQueue (0.00s)
=== RUN   TestFifoQueue_GetExist_Statistic
    queue_test.go:72: Failed drop queue: Read access to space '_trigger' is denied for user 'test' (0x2a)
--- FAIL: TestFifoQueue_GetExist_Statistic (0.00s)
=== RUN   TestFifoQueue_Put
    queue_test.go:124: Failed drop queue: Read access to space '_trigger' is denied for user 'test' (0x2a)
--- FAIL: TestFifoQueue_Put (0.00s)
=== RUN   TestFifoQueue_Take
    queue_test.go:167: Failed drop queue: Read access to space '_trigger' is denied for user 'test' (0x2a)
--- FAIL: TestFifoQueue_Take (0.00s)
=== RUN   TestFifoQueue_TakeTyped
    queue_test.go:299: Task data after take not equal with example. &queue_test.customData{customField:""} != &queue_test.customData{customField:"put_data"}
    queue_test.go:302: Task data after take not equal with example. &queue_test.customData{customField:""} != &queue_test.customData{customField:"put_data"}
    queue_test.go:263: Failed drop queue: Read access to space '_trigger' is denied for user 'test' (0x2a)

```

Reply:
Sorry, it seems that you're correct and my verification with removing `_trigger` grants were wrong. Seems like I ran tests without cleaning cache.

Nonetheless, I think creating `_trigger` if it not exists is a bad pattern. Please, use
```lua
if box.space._trigger ~= nil then
    box.schema.user.grant('test', 'read', 'space', '_trigger')
end
```
for `_trigger` and use the same for all tarantool internal spaces (`_fk_constraint`, `_ck_constraint`, `_func_index`).

Reply:
You are right, that looks better. Fixed.

Reply:
The constraint `Tarantool != 1.10` is also removed in reusable_testing.